native/dlarfb.go

```diff
@@ -0,0 +1,341 @@
+// Copyright ©2015 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package native
+
+import (
+	"github.com/gonum/blas"
+	"github.com/gonum/blas/blas64"
+	"github.com/gonum/lapack"
+)
+
+// Dlarfb applies a block reflector to a matrix.
+//
+// In the call to Dlarfb, c is multiplied by the implicitly defined matrix h.
+//  c = h * c if side == Left and trans == NoTrans
+//  c = c * h if side == Right and trans == NoTrans
+//  c = h^T * c if side == Left and trans == Trans
+//  c = c * h^t if side == Right and trans == NoTrans
+// h is a product of elementary reflectors. direct sets the direction of multiplication
+//  h = h_1 * h_2 * ... * h_k if direct == Forward
+//  h = h_k * h_k-1 * ... * h_1 if direct == Backward
+// The combination of direct and store defines the orientation of the elementary
+// reflectors. In all cases the ones on the diagonal are implicitly represented.
+//
+//  If direct == lapack.Forward and store == lapack.Columnwise
+//  V = (  1       )
+//      ( v1  1    )
+//      ( v1 v2  1 )
+//      ( v1 v2 v3 )
+//      ( v1 v2 v3 )
+//  If direct == lapack.Forward and store == lapack.Rowwise
```

Comment:
Done.

Maybe we should work on eliminating hyphenated words. I'm not looking forward to camel rule.